src/app/state/tetris/tetris.state.ts

```diff
@@ -0,0 +1,130 @@
+import { PieceFactory } from '@angular-tetris/factory/piece-factory';
+import { GameState } from '@angular-tetris/interface/game-state';
+import { Piece } from '@angular-tetris/interface/piece/piece';
+import { Speed } from '@angular-tetris/interface/speed';
+import { Tile } from '@angular-tetris/interface/tile/tile';
+import { MatrixUtil } from '@angular-tetris/interface/utils/matrix';
+import { LocalStorageService } from '@angular-tetris/services/local-storage.service';
+import { DestroyRef, Injectable, computed, effect, inject, signal } from '@angular/core';
+
+export interface TetrisState {
+  matrix: Tile[];
+  current: Piece | null;
+  next: Piece;
+  hold: Piece;
+  canHold: boolean;
+  points: number;
+  locked: boolean;
+  sound: boolean;
+  initSpeed: Speed;
+  speed: Speed;
+  initLine: number;
+  clearedLines: number;
+  gameState: GameState;
+  saved: TetrisState;
+  max: number;
+}
+
+const createInitialState = (pieceFactory: PieceFactory): TetrisState => ({
+  matrix: MatrixUtil.getStartBoard(),
+  current: null,
+  next: pieceFactory.getRandomPiece(),
+  hold: pieceFactory.getNonePiece(),
+  canHold: true,
+  points: 0,
+  locked: true,
+  sound: true,
+  initLine: 0,
+  clearedLines: 0,
+  initSpeed: 1,
+  speed: 1,
+  gameState: GameState.Loading,
+  saved: null,
+  max: LocalStorageService.maxPoint
+});
+
+const isObjectShallowEqual = (a: any, b: any) => a == b;
```

Comment:
@mobi1025 what happens if we use triple equal here? Apparently both `==` and `===` yield the same result for `object` comparison. The triple equal (`===`) is technically faster because it can skip the type conversion step before actually comparing the objects. 

See the `IsLooselyEqual` and `IsStrictlyEqual` below 

- https://tc39.es/ecma262/multipage/abstract-operations.html#sec-islooselyequal
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

Reply:
TIL sir I thought the `==` and `===` performance are the same so I use `==` instead 😆